File: Websites/Codex5.3_VistaCanvas/backend/app/schemas/inspiration.py
from pydantic import BaseModel


class ApodItem(BaseModel):
    title: str
    explanation: str
    image_url: str | None
    page_url: str
    date: str
    copyright: str | None
    source: str = "NASA Astronomy Picture of the Day"


class ScenicWeather(BaseModel):
    location: str
    temperature_c: float
    weather_label: str
    wind_speed_kmh: float
    sunrise: str | None = None
    sunset: str | None = None


class NatureQuote(BaseModel):
    content: str
    author: str
    source: str = "ZenQuotes"


class InspirationFeed(BaseModel):
    apod: ApodItem | None
    scenic_weather: list[ScenicWeather]
    quote: NatureQuote | None = None
    apod_error: str | None = None
    weather_error: str | None = None
    quote_error: str | None = None

File: Websites/Codex5.3_VistaCanvas/frontend/src/types/inspiration.ts
export interface ApodItem {
  title: string;
  explanation: string;
  image_url: string | null;
  page_url: string;
  date: string;
  copyright: string | null;
  source: string;
}

export interface ScenicWeather {
  location: string;
  temperature_c: number;
  weather_label: string;
  wind_speed_kmh: number;
  sunrise: string | null;
  sunset: string | null;
}

export interface NatureQuote {
  content: string;
  author: string;
  source: string;
}

export interface InspirationFeed {
  apod: ApodItem | null;
  scenic_weather: ScenicWeather[];
  quote: NatureQuote | null;
  apod_error: string | null;
  weather_error: string | null;
  quote_error: string | null;
}
